lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -439,3 +451,108 @@ getPoolPerformanceEstimate sp dl rp pid = do
             / fromIntegral (unCoin $ totalStake rp)
             -- _poolHistoryActiveSize would be incorrect here
         }
+
+-- | Retrieve a 'StakePoolsSummary'.
+--
+-- TODO: This summary is from the "Cardano.Pool.Rank" module;
+-- the one from the "Cardano.Wallet.Primitive.Types" will be
+-- removed.
+--
+-- TODO: For reasons of performance, this summary does not
+-- fill in the 'performanceEstimate' field of each stake pool.
+-- As it is not needed for the new reward computation,
+-- it will be removed in the future.
+getStakePoolsSummary'
+    :: BF.MonadBlockfrost m
+    => Tracer m Log
+    -> m StakePoolsSummary
+getStakePoolsSummary' tr = do
+    let mkPoolMap pids = Map.fromList [ (fromPoolId p,p) | p <- pids ]
+    pools <- mkPoolMap <$> getAllPages BF.listPools'
+
+    rp <- getRewardParams
+    let getPool pid = do
+            info <- fromPoolInfo rp <$> BF.getPool pid
+            traceWith tr $ MsgRewardInfoPool (fromPoolId pid) info
+            pure info
+
+    StakePoolsSummary rp <$> mapM getPool pools
+
+getRewardParams :: BF.MonadBlockfrost m => m RewardParams
+getRewardParams = do
+    BF.ProtocolParams{..} <- BF.getLatestEpochProtocolParams
+    BF.Network{..} <- BF.getNetworkInfo
+    BF.EpochInfo{..} <- BF.getLatestEpoch
+    let BF.NetworkSupply{..} = _networkSupply
+        eta
+            | _protocolParamsDecentralisationParam >= 0.8 = 1
+            | otherwise = 1 -- wrong value
+                -- NOTE: The correct value is  blocksMade % expectedBlocks
+                -- But since we are long after d = 1, we ignore that here.
+        deltaR1 = rationalToCoinViaFloor $
+            min 1 eta
+                * realToFrac _protocolParamsRho
+                * fromIntegral _supplyReserves
+        rationalToCoinViaFloor :: Rational -> Coin
+        rationalToCoinViaFloor = Coin . floor
+        rPot = fromLovelaces _epochInfoFees <> deltaR1
+        rp = RewardParams
+            { nOpt =
+                fromIntegral _protocolParamsNOpt
+            , a0 =
+                realToFrac _protocolParamsA0
+            , r =
+                rPot
+            , totalStake =
+                fromLovelaces _supplyTotal
+            }
+    pure rp
```

Comment:
Stylistic nit:
```suggestion
        rp = 
    pure RewardParams
        { nOpt = fromIntegral _protocolParamsNOpt
        , a0 = realToFrac _protocolParamsA0
        , r = rPot
        , totalStake = fromLovelaces _supplyTotal
        }
```
(I'd write it this way to avoid introducing `rp` as a short-lived local name)